### PR TITLE
py-kubernetes: add new package

### PR DESCRIPTION
**Pull request by vvolkl**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-kubernetes/package.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    """Official Python client library for kubernetes. """
+
+    homepage = "https://kubernetes.io"
+    git      = "https://github.com/kubernetes-client/python.git"
+    pypi = "kubernetes/kubernetes-17.17.0.tar.gz"
+
+    maintainers = ['vvolkl']
+
+    version('18.17.0a1', sha256='5746f5aeacdd2bd7e5ec2250db2f3ca4008a0c4f26e1b6e444e7f3d663f3af6a')
+    version('17.17.0b1', sha256='2a5cd1b80f821e7b1b4e120bb306fc0ab6e779450df2554f0d8199fcb389f63b')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-kubernetes/package.py]: Is there any reason to add alpha and beta versions? If not, I would remove them. If you need them, you should mark 17.17.0 as preferred=True.

File: var/spack/repos/builtin/packages/py-kubernetes/package.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    version('10.0.1',    sha256='3770a496663396ad1def665eeadb947b3f45217a08b64b10c01a57e981ac8592')
+    version('9.0.0', sha256='a8b0aed55ba946faea660712595a52ae53a8854df773d96f47a63fa0c9d4e3bf')
+
+    depends_on('py-certifi@14.05.14:')
+    depends_on('py-six@1.9.0:')
+    depends_on('py-python-dateutil@2.5.3:')
+    depends_on('py-setuptools@21.0.0:', type=('build'))
+    depends_on('py-pyyaml@3.12:')
+    depends_on('py-google-auth@1.0.1:')
+    depends_on('py-ipaddress@1.0.17:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-kubernetes/package.py]:
```suggestion
    depends_on('py-ipaddress@1.0.17:', when='^python@:2.8', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-kubernetes/package.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    depends_on('py-websocket-client@0.43:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-kubernetes/package.py]:
```suggestion
    depends_on('py-websocket-client@0.32:0.39,0.43:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-kubernetes/package.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    depends_on('py-requests')
+    depends_on('py-requests-oauthlib')
+    depends_on('py-urllib3@1.24.2:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-kubernetes/package.py]: Other than setuptools, these should all be `type=('build', 'run')`

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-kubernetes/package.py (hunk 1 above)
Comment (vvolkl) [reply var/spack/repos/builtin/packages/py-kubernetes/package.py]: No, not really. I removed them.

File: var/spack/repos/builtin/packages/py-kubernetes/package.py (hunk 4 above)
Comment (vvolkl) [reply var/spack/repos/builtin/packages/py-kubernetes/package.py]: done!

**Comment by vvolkl** [comment]
@adamjstewart thanks for the comments! should be fine now

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**